src/tag-editor/__tests__/get-tags-diff.test.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+
+import { TagEditorProps } from '../../../lib/components/tag-editor';
+import { getTagsDiff } from '../utils';
+
+jest.mock('@cloudscape-design/component-toolkit/internal', () => ({
+  warnOnce: jest.fn(),
+}));
```

Comment:
This mocks too much, all of the toolkit internals. Even if your tests pass, you make live for future devs editing this code harder.

The intended setup for testing warnings is here: https://github.com/cloudscape-design/components/blob/7f14eff7001437b54eeeca2ccf6c165afc56313d/src/app-layout/__tests__/multi-layout.test.tsx#L192-L199